libs/basics/src/iterators/types.ts

```diff
@@ -95,6 +95,24 @@ export interface IteratorPlus<T> extends Iterable<T> {
    */
   last(): T | undefined;
 
+  /**
+   * Pipes elements from `this` by collecting them together into a string until
+   * a full line is formed, then yielding that line including the newline if
+   * any.
+   *
+   * @example
+   *
+   * ```ts
+   * const input = ['hello', ' world', '!', '\n', 'goodbye', 'world'];
+   *
+   * expect(iter(input).lines().toArray()).toEqual([
+   *   'hello world!\n',
+   *   'goodbye world'
+   * ]);
+   * ```
+   */
+  lines(): IteratorPlus<T extends string ? string : unknown>;
```

Comment:
Feels a little weird to have a string-specific method on `iter`, which is otherwise oblivious to the contents of the iterable. Since this has such a specific use case (converting a file stream to an iterable of lines), maybe it should just be a standalone function?

Reply:
Ah, I forgot about `sum`!